web/src/copy_and_paste.ts

```diff
@@ -385,15 +385,16 @@ export function is_white_space_pre(paste_html: string): boolean {
     );
 }
 
-function is_single_image(paste_html: string): boolean {
+export function is_single_image(paste_html: string): boolean {
     const html_fragment = new DOMParser()
         .parseFromString(paste_html, "text/html")
         .querySelector("body");
     assert(html_fragment !== null);
     return (
-        html_fragment.childNodes.length === 1 &&
-        html_fragment.firstElementChild !== null &&
-        html_fragment.firstElementChild.nodeName === "IMG"
+        paste_html.includes("urn:schemas-microsoft-com:office") ||
```

Comment:
This check is too imprecise; text containing that anywhere in a code block should not trigger this behavior.

You need to use the HTML parse to do this check.